widgetssdk/src/main/java/com/glia/widgets/entrywidget/EntryWidgetView.kt

```diff
@@ -0,0 +1,44 @@
+package com.glia.widgets.entrywidget
+
+import android.content.Context
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.glia.widgets.R
+import com.glia.widgets.di.Dependencies
+import com.glia.widgets.entrywidget.adapter.EntryWidgetAdapter
+import com.glia.widgets.entrywidget.adapter.EntryWidgetItemDecoration
+import com.glia.widgets.helper.getDrawableCompat
+
+internal class EntryWidgetView(
+    context: Context
+) : RecyclerView(context, null, 0), EntryWidgetContract.View {
+    var onDismissListener: (() -> Unit)? = null
+
+    private lateinit var controller: EntryWidgetContract.Controller
+    private val adapter = EntryWidgetAdapter().also { setAdapter(it) }
+
+    init {
+        layoutManager = LinearLayoutManager(context)
+        getDrawableCompat(R.drawable.bg_separator_gray)?.let {
+            addItemDecoration(EntryWidgetItemDecoration(it))
+        }
+        adapter.onItemClickListener = {
+            controller.onItemClicked(it)
+        }
+
+        setController(Dependencies.controllerFactory.entryWidgetController)
```

Comment:
Does it make sense to call `setController` `before adapter.onItemClickListener` to prevent any chance of `controller.onItemClicked` call before controller is set?

Reply:
But without the controller, there is nothing to display. So, no click events should be before `setController`.